### PR TITLE
Support hybrid search in Aggregate

### LINKED ISSUE #2482: (bug/feature request) Hybrid operator not compatible with aggregations
HI,

It looks like aggregations are not supported with hybrid search.

Search:
```{
  search: Get {
    Test1(
      limit: 20
      offset: 0
      hybrid: {query: "helo", alpha: 0.750000}
    ) {
      wpsolr_id
      wpsolr_title
    }
  }
  search_count: Aggregate {
    Test1(
      hybrid: {query: "helo", alpha: 0.750000}
    ) {
      meta {
        count
      }
    }
  }
}
```

gives:
```
{
  "errors": [
    {
      "locations": [
        {
          "column": 7,
          "line": 14
        }
      ],
      "message": "Unknown argument \"hybrid\" on field \"Test1\" of type \"AggregateObjectsObj\".",
      "path": null
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by parkerduckworth**
### What's being changed:

The existing hybrid search functionality was hardcoded to work only with `Get{}` queries. This PR generalizes the hybrid search code, and uses it to support hybrid search within `Aggregate{}` queries, for use with filters _and_ grouping.

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [x] Chaos pipeline run or not necessary. 
    - Link to pipeline: https://github.com/semi-technologies/weaviate-chaos-engineering/actions/runs/3888090623
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

Closes #2482

### CODE REVIEW

**Comment by parkerduckworth** [comment]
@etiennedi yes, I think we should target `stable/v1.17` to get it out earlier

File: adapters/repos/db/aggregator/filtered.go [hunk 1]
```diff
@@ -35,49 +37,105 @@ func newFilteredAggregator(agg *Aggregator) *filteredAggregator {
...
-			return nil, errors.Wrap(err, "retrieve doc IDs from searcher")
+			return nil, nil, fmt.Errorf("aggregate sparse search: %w", err)
 		}
+
+		return sparse, dists, nil
+	}
+
+	denseSearch := func(vec []float32) ([]*storobj.Object, []float32, error) {
+		if vec == nil {
+			return nil, nil, fmt.Errorf("must provide hybrid search vector if alpha > 0")
```
Comment (donomii) [inline adapters/repos/db/aggregator/filtered.go:69]: I'm not too sure about exactly what Aggregate{} does, but in Get{}, the vector is optional and if not present, is calculated from the query parameter.  Is it possible to do the same thing here?

File: entities/storobj/storage_object.go [hunk 2]
```diff
@@ -325,14 +374,13 @@ func SearchResults(in []*Object, additional additional.Properties) search.Result
 	return out
 }
 
-func SearchResultsWithDists(in []*Object, additional additional.Properties,
+func SearchResultsWithDists(in []*Object, addl additional.Properties,
```
Comment (donomii) [inline entities/storobj/storage_object.go:377]: I don't see the benefit of doing this

File: adapters/repos/db/aggregator/filtered.go (hunk 1 above)
Comment (parkerduckworth) [reply adapters/repos/db/aggregator/filtered.go:69]: This is actually still the case, although the logic exists outside the local usecase closure, and lives instead in the generic hybrid search logic ([see here](https://github.com/semi-technologies/weaviate/blob/feature/agg-hybrid-support/usecases/traverser/hybrid/searcher.go#L266-L284))

But currently, aggregation does not support any module providers, so this feature is unused by Aggregation until we decide to add support for modules in Aggregation queries

File: entities/storobj/storage_object.go (hunk 2 above)
Comment (parkerduckworth) [reply entities/storobj/storage_object.go:377]: The original name collides with the imported package and broke compilation, so I went with this common abbreviation  instead

**Comment by eostis** [comment]
@parkerduckworth I made a quick test with the latest master version according to https://weaviate.io/developers/weaviate/current/installation/#run-an-unreleased-version

I got the following error:
image
image
image
image
image
image

**Comment by parkerduckworth** [comment]
@eostis currently, a search vector is required for aggregation, since the aggregator has no access to the modules. I will look into providing the aggregator access to the module system. But in the meantime, is providing a search vector a reasonable workaround for your use case?

**Comment by eostis** [comment]
@parkerduckworth For e-commerce, aggregation should be able to work the same for Where, NearText, or hybrid. I'll wait for the fix. Do you want me to create a new issue ?

**Comment by parkerduckworth** [comment]
@eostis I just made a fix that will allow the query from your example above. When the CI pipeline is complete, I will send you an image tag so you can try it out

**Comment by parkerduckworth** [comment]
@eostis try this:
```
semitechnologies/weaviate:preview-support-implicit-search-vectors-for-aggregation-queries-518a5fd
```

**Comment by eostis** [comment]
@parkerduckworth Looks good now !

image

image

**Pull request merged**